index.js

```diff
@@ -0,0 +1,108 @@
+    .min(0);
+const SCHEMA_EMAIL = Joi.alternatives().try(
+    Joi.object().keys({ addresses: SCHEMA_ADDRESSES, statuses: SCHEMA_STATUSES }),
+    SCHEMA_ADDRESS
+    );
+const SCHEMA_BUILD_SETTINGS = Joi.object()
+    .keys({
+        email: SCHEMA_EMAIL
+    }).unknown(true);
+const SCHEMA_BUILD_DATA = Joi.object()
+    .keys({
+        settings: SCHEMA_BUILD_SETTINGS,
+        status: SCHEMA_STATUS
+    });
+const SCHEMA_SMTP_CONFIG = Joi.object()
+    .keys({
+        host: Joi.string(),
+        port: Joi.number().integer(),
+        from: SCHEMA_ADDRESS
+    });
+
+const DESCRIPTION_MAP = {
+    SUCCESS: 'Everything looks good!',
+    FAILURE: 'Did not work as expected.',
+    ABORTED: 'Aborted mid-flight',
+    RUNNING: 'Testing your code...',
+    QUEUED: 'Looking for a place to park...'
+};
+const DEFAULT_STATUSES = ['FAILURE'];
+
+class EmailNotifier {
+    /**
+    * Constructs an EmailNotifier
+    * @constructor
+    * @param {object} config - Screwdriver config object initialized in API
+    * @param {Hapi.server} server - server initialized in API
+    * @param {string} eventName - name of notification event to listen on
+    */
+    constructor(config, server, eventName) {
+        this.config = Joi.attempt(config, SCHEMA_SMTP_CONFIG,
+            'Invalid config for email notifications');
+        this.server = server;
+        this.eventName = eventName; // The event that notify() will trigger a listener on
+    }
+
+    /**
+    * Sets listener on server event of name 'eventName'
+    * Currently, event is triggered with a build status is updated
+    * @method notify
+    * @return {Promise} resolves to false if status is not in notification statuses
+    *                   resolves to emailer if status is in notification statuses
+    */
+    notify() {
+        return new Promise((resolve, reject) => {
+            this.server.on(this.eventName, (buildData) => {
+                // Check buildData format against SCHEMA_BUILD_DATA
+                try {
+                    Joi.attempt(buildData, SCHEMA_BUILD_DATA, 'Invalid build data format');
+                } catch (e) {
+                    return reject(e);
+                }
+
+                if (typeof buildData.settings.email === 'string' ||
+                    buildData.settings.email instanceof String) {
+                    buildData.settings.email = {
+                        addresses: buildData.settings.email,
+                        statuses: DEFAULT_STATUSES
+                    };
+                }
+
+                if (buildData.settings.email.statuses.indexOf(buildData.status) <= -1) {
+                    return resolve(null);
+                }
+
+                const message = DESCRIPTION_MAP[buildData.status];
+
+                const mailOpts = {
+                    from: this.config.from,
+                    to: buildData.settings.email.addresses,
+                    subject: `Screwdriver Build ${buildData.status}`,
```

Comment:
Maybe API should also send Build number. Otherwise this msg doesn't really help.